meson.build

```diff
@@ -0,0 +1,222 @@
+if harfbuzz_dep.found()
+    deps += harfbuzz_dep
+    conf.set('CONFIG_HARFBUZZ', 1)
+endif
+
+req_fontconfig = get_option('fontconfig')
+
+if host_machine.system() != 'windows' and host_machine.system() != 'darwin'
+    req_fontconfig = get_option('fontconfig').disabled()
+endif
+
+fontconfig_dep = dependency('fontconfig', version: '>= 2.10.92', required: req_fontconfig)
+fontconfig = false
+if fontconfig_dep.found()
+    deps += fontconfig_dep
+    conf.set('CONFIG_FONTCONFIG', 1)
+    fontconfig = true
+endif
+
+if get_option('test')
+    deps += dependency('libpng', version: '>= 1.2.0')
+    conf.set('CONFIG_LIBPNG', 1)
+endif
+
+# DirectWrite
+directwrite = false
+if host_machine.system() == 'windows' and not get_option('directwrite').disabled()
+    if cc.has_header('windows.h')
+        deps += cc.find_library('dwrite', required: true)
+        conf.set('CONFIG_DIRECTWRITE', 1)
+        directwrite = true
+    elif get_option('directwrite').enabled()
+        error('DirectWrite was enabled explicitly, but required header is missing.')
+    endif
+endif
+
+# CoreText
+coretext = false
+if host_machine.system() == 'darwin' and not get_option('coretext').disabled()
+    coretext_dep = dependency('appleframeworks', modules: ['CoreText', 'CoreFoundation'], required: false)
+    if cc.has_type('CTRunRef', prefix: '#include <CoreText/CoreText.h>', dependencies: coretext_dep)
+        deps += coretext_dep
+        conf.set('CONFIG_CORETEXT', 1)
+        coretext = true
+    elif get_option('coretext').enabled()
+        error('CoreText was enabled explicitly, but required headers or frameworks are missing.')
+    endif
+endif
+
+if get_option('require-system-font-provider') and not fontconfig and not directwrite and not coretext
+    error('''Either DirectWrite (on Windows), CoreText (on OSX), or Fontconfig (Linux, other) is required.
+If you really want to compile without a system font provider, set -Drequire-system-font-provider=false''')
+endif
+
+# dirent.h
+if host_machine.system() == 'windows' and not cc.has_header('dirent.h')
+    deps += subproject('tronkko-dirent').get_variable('tronkko_dirent_dep')
+endif
+
+foreach name: check_functions
+    # workaround for https://github.com/mesonbuild/meson/issues/3672
+    if cc.has_function(name) and cc.has_header_symbol('string.h', name)
+        conf.set('HAVE_@0@'.format(name.to_upper()), 1)
+    endif
+endforeach
+
+conf.set('PACKAGE_NAME', 'libass')
+conf.set('PACKAGE_VERSION', meson.project_version())
+
+config_h = configure_file(output: 'config.h', configuration: conf)
+
+incs = include_directories('.', 'libass')
+
+subdir('libass')
+if get_option('test')
+    subdir('test')
+endif
+if get_option('profile')
+    subdir('profile')
+endif
+
+# libass.pc
+pkg = import('pkgconfig')
+pkg.generate(libass,
+             name: 'libass',
+             description: 'LibASS is an SSA/ASS subtitles rendering library',
+             subdirs: meson.project_name())
```

Comment:
Not sure about the install location for the headers; I think I'd have to just test. I paid relatively little attention to Linux past testing when I initially PR'd, since my primary use case is static linking on Windows/OSX.

No clue what's up with the versions. I have the minimum harfbuzz version listed here as `>= 0.9.5` and freetype is... weird.

Reply:
Versions should be:
```
fontconfig >= 2.10.92, harfbuzz >= 1.2.3, fribidi >= 0.19.0, freetype2 >= 9.10.3
```
Min Harfbuzz version was bumped in 7fc053fde7455fe8faacd0f97bd738f2d0a1076a.

Reply:
Fixed the harfbuzz version. Still not sure about the install location/include dirs, I'll have to boot up my ubuntu VM.

Reply:
The install location/include dirs should be good; only remaining issue is freetype.